Realm/Realm/Sync/App.cs

```diff
@@ -158,7 +158,7 @@ public static App Create(AppConfiguration config)
             var nativeConfig = new Native.AppConfiguration
             {
                 AppId = config.AppId,
-                BaseFilePath = config.BaseFilePath ?? InteropConfig.DefaultStorageFolder,
+                BaseFilePath = config.BaseFilePath ?? InteropConfig.GetDefaultStorageFolder("app configuration"),
```

Comment:
This would expand to

```
Couldn't determine a writable folder where to store app configuration file. Specify absolute path manually.
```

This is unfortunately very vague as 1) we don't really store app configurations and 2) it doesn't instruct the user how to specify the absolute path. While I like that we're trying to reuse some of the wording around these exception messages, I think in this case it's better to allow the consumer to specify the complete error message for maximum flexibility. Then we can supply something like:

```
Couldn't determine a writable folder to store app files (such as metadata and Realm files). When constructing the app, set AppConfiguration.BaseFilePath to an absolute path where the app is allowed to write.
```